web/pdf_rendering_queue.js

```diff
@@ -126,10 +126,30 @@ class PDFRenderingQueue {
         return view;
       }
     }
+    const firstId = visible.first.id,
+      lastId = visible.last.id;
+
+    // All the visible views have rendered; try to handle any "holes" in the
+    // page layout (can happen e.g. with spreadModes at higher zoom levels).
+    if (lastId - firstId > 1) {
+      const holeIds = new Set();
+      for (let i = 0, ii = lastId - firstId; i <= ii; i++) {
+        holeIds.add(scrolledDown ? firstId + i : lastId - i);
+      }
+      for (let i = 0; i < numVisible; i++) {
+        holeIds.delete(visibleViews[i].id);
+      }
+      for (const id of holeIds) {
+        const holeView = views[id - 1];
```

Comment:
Thinking about this some more, the multiple loops approach doesn't feel all that great.
Another possible approach would be the following, which uses the fact that already rendered pages will be skipped:
```javascript
      for (let i = 0, ii = lastId - firstId; i <= ii; i++) {
        const holeId = scrolledDown ? firstId + i : lastId - i,
          holeView = views[holeId - 1];
        if (holeView && !this.isViewFinished(holeView)) {
          return holeView;
        }
      }
```